src/LibraryManager.Vsix/UI/Models/InstallDialogViewModel.cs

```diff
@@ -372,29 +388,186 @@ private static bool IsAnyFileSelected(IReadOnlyList<PackageItem> children)
 
-                manifest.AddLibrary(new LibraryInstallationState
+                LibraryInstallationState libraryInstallationState = new LibraryInstallationState
                 {
                     LibraryId = PackageId,
                     ProviderId = selectedPackage.ProviderId,
                     DestinationPath = InstallationFolder.DestinationFolder,
                     Files = SelectedFiles.ToList()
-                });
-
-                await manifest.SaveAsync(_configFileName, CancellationToken.None).ConfigureAwait(false);
+                };
 
                 EnvDTE.Project project = VsHelpers.DTE.SelectedItems.Item(1)?.ProjectItem?.ContainingProject;
                 project?.AddFileToProjectAsync(_configFileName);
 
-                await _libraryCommandService.RestoreAsync(_configFileName, CancellationToken.None).ConfigureAwait(false);
+                await Shell.ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                RunningDocumentTable rdt = new RunningDocumentTable(Shell.ServiceProvider.GlobalProvider);
+                IVsTextBuffer document = rdt.FindDocument(Path.GetFullPath(_configFileName)) as IVsTextBuffer;
+
+                // The file isn't open. So we'll write to disk directly
+                if (document == null)
+                {
+                    manifest.AddLibrary(libraryInstallationState);
+
+                    await manifest.SaveAsync(_configFileName, CancellationToken.None).ConfigureAwait(false);
+
+                    await _libraryCommandService.RestoreAsync(_configFileName, CancellationToken.None).ConfigureAwait(false);
 
-                _dispatcher.Invoke(() =>
+                    _dispatcher.Invoke(() =>
+                    {
+                        _closeDialog(true);
+                    });
+                }
+                else
                 {
-                    _closeDialog(true);
-                });
+                    // libman.json file is open, so we will write to the textBuffer.
+                    _dispatcher.Invoke(() =>
+                    {
+                        _closeDialog(true);
+                    });
+
+                    InsertIntoTextBuffer(document, libraryInstallationState);
+
+                    // Save manifest file so we can restore library files.
+                    rdt.SaveFileIfDirty(Path.GetFullPath(_configFileName));
+                }
             }
             catch { }
         }
 
+        private void InsertIntoTextBuffer(IVsTextBuffer document, LibraryInstallationState libraryInstallationState)
+        {
+            ITextBuffer textBuffer = GetTextBuffer(document);
+
+            if (textBuffer != null)
+            {
+                string insertionText;
+
+                JSONMember libraries = GetLibraries(textBuffer);
+
+                if (libraries != null && libraries.Children != null)
+                {
+                    JSONArray jsonArray = libraries.Children.OfType<JSONArray>().First();
```

Comment:
>First [](start = 81, length = 5)

Was this supposed to be FirstOrDefault or is the if(jsonArray != null) not necessary?

Reply:
Currently we only have one JSONArray for libraries and its a required property. So First should be fine. Null check for jsonArray might not be required.

---
In reply to: [197855633](https://github.com/aspnet/LibraryManager/pull/246#discussion_r197855633) [](ancestors = 197855633)

Reply:
Yeah, the null check doesn't do anything here as the .First() call will throw if there isn't anything

---
In reply to: [197926691](https://github.com/aspnet/LibraryManager/pull/246#discussion_r197926691) [](ancestors = 197926691,197855633)

Reply:
Updated in version 8.

---
In reply to: [197927099](https://github.com/aspnet/LibraryManager/pull/246#discussion_r197927099) [](ancestors = 197927099,197926691,197855633)